pages/events/[id].js

```diff
@@ -0,0 +1,48 @@
+import { useRouter } from "next/router";
+import styled from "styled-components";
+
+const StyledList = styled.ul`
+  list-style: none;
+  padding: 0;
+  margin: 0;
+`;
+const StyledListItem = styled.li`
+  text-align: center;
+  margin: 60px;
+`;
+
+export default function EventDetails({ events }) {
+  const router = useRouter();
+  const { id } = router.query;
+
+  const event = events.find((event) => {
+    return event.id === parseInt(id);
```

Comment:
if you use the method described above ids will be strings, so you won't need the parseInt
```suggestion
    return event.id === id;
```